RevenueCatUI/Data/TemplateViewConfiguration.swift

```diff
@@ -236,54 +236,56 @@ extension TemplateViewConfiguration.PackageConfiguration {
             }
 
         case let .multiTier(tiers, localization):
-            let allTiers: [PaywallData.Tier: (package: MultiPackage, tierName: String)] = try Dictionary(
-                tiers.compactMap { tier in
-                    guard let localization = localization[tier.id] else {
-                        throw TemplateError.missingLocalization(tier)
-                    }
-
-                    let tierName = localization.tierName ?? ""
-
-                    let filteredPackages = Self.processPackages(
-                        from: packages,
-                        filter: tier.packages,
-                        activelySubscribedProductIdentifiers: activelySubscribedProductIdentifiers,
-                        localization: localization,
-                        locale: locale,
-                        showZeroDecimalPlacePrices: showZeroDecimalPlacePrices
-                    )
+            let filteredTiers: [(PaywallData.Tier, (MultiPackage, String))] = try tiers.compactMap { tier in
```

Comment:
I recommend hiding whitespace when reviewing this file. 

Instead of filtering the `tiers` inline while creating the `Dictionary`, it is now filtering the `tiers` separately first. Then we can use the filtered list to grab the first item, for our `firstTier`. Then the filtered list is passed into the `Dictionary` constructor as before.